### PR TITLE
Remove units from client.

### PR DESCRIPTION
**Pull request by vjkoskela**

### CODE REVIEW

File: src/main/java/com/arpnetworking/metrics/impl/TsdMetrics.java [hunk 1]
```diff
@@ -342,6 +321,14 @@ private boolean assertIsOpen(final boolean isOpen) {
         return isOpen;
     }
 
+    static double convertTimeUnit(final double valueFrom, final TimeUnit unitFrom, final TimeUnit unitTo) {
+        if (unitFrom.ordinal() < unitTo.ordinal()) {
```
Comment (speezepearson) [inline src/main/java/com/arpnetworking/metrics/impl/TsdMetrics.java]: Relying on the enum ordering seems pretty wonky. Could we instead
```java
double toPerFrom = unitTo.convert(1, unitFrom)
if (toPerFrom == 0) {
    toPerFrom = 1 / unitFrom.convert(1, unitTo)
}
return valueFrom * multiplicativeFactor
final double result = valueFrom * toPerFrom
```
or something semantically equivalent?

File: src/test/java/com/arpnetworking/metrics/impl/TsdTimerTest.java [hunk 2]
```diff
@@ -166,8 +168,7 @@ public void testGetElapsedAfterStop() {
         Mockito.verify(logger, Mockito.never()).warn(MockitoHamcrest.argThat(Matchers.any(String.class)));
         timer.getValue();
         Mockito.verify(logger).warn(MockitoHamcrest.argThat(Matchers.any(String.class)));
-        timer.getUnit();
-        Mockito.verify(logger, Mockito.times(2)).warn(MockitoHamcrest.argThat(Matchers.any(String.class)));
+        Mockito.verify(logger, Mockito.times(1)).warn(MockitoHamcrest.argThat(Matchers.any(String.class)));
```
Comment (speezepearson) [inline src/test/java/com/arpnetworking/metrics/impl/TsdTimerTest.java]: [nit] This looks like a restatement (or slightly stronger version of) the previous line?

File: src/main/java/com/arpnetworking/metrics/impl/TsdMetrics.java (hunk 1 above)
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/metrics/impl/TsdMetrics.java]: Yeah, that's definitely better.

File: src/test/java/com/arpnetworking/metrics/impl/TsdTimerTest.java (hunk 2 above)
Comment (vjkoskela) [reply src/test/java/com/arpnetworking/metrics/impl/TsdTimerTest.java]: That was left over statement when removing the call to `getUnit`; both would have emitted a warning.

**Pull request merged**